ApiConnection.cs

```diff
@@ -0,0 +1,144 @@
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using eWayCRM.API;
+using MigrateDocuments.Configuration;
+
+namespace MigrateDocuments
+{
+    class ApiConnection
+    {
+        private readonly Connection connection;
+
+        public ApiConnection(RootConfigSection config)
+        {
+            string[] appVersionElements = config.AssemblyVersion.ToString().Split('.');
+
+            connection = new Connection(
+                config.Connection.Url,
+                config.Connection.UserName,
+                config.Connection.Password,
+                $"{config.AssemblyName}_v{appVersionElements[0]}.{appVersionElements[1]}"
+            );
+
+            connection.EnsureLogin();
+        }
+
+        public JObject GetFolder(string folderName)
+        {
+            return connection.CallMethod($"Get{folderName}");
+        }
+
+        public JObject SearchFolder(string folderName, JObject transmitObject)
+        {
+            return connection.CallMethod($"Search{folderName}", JObject.FromObject(new
+            {
+                transmitObject
+            }));
+        }
+
+        public IEnumerable<Guid> GetDocuments(Guid itemGuid, string folderName)
+        {
+            return connection.GetItemsByItemGuids($"Get{folderName}ByItemGuids", new Guid[] { itemGuid }, false, true)
+                .Single()["Relations"]
+                .Where(x => x.Value<string>("ForeignFolderName") == "Documents")
+                .Select(x => new Guid(x.Value<string>("ForeignItemGUID")))
+                .ToArray();
+        }
+
+        public void DownloadDocument(Guid documentGuid, string directory)
+        {
+            var revisionInfo = connection.GetLatestRevision(documentGuid)["Datum"];
+            string documentName = revisionInfo.Value<string>("FileAs");
+            string filePath = Path.Combine(directory, documentName);
+
+            if (File.Exists(filePath))
+            {
+                Logger.LogDebug($"File '{filePath}' already exist");
+                return;
+            }
+
+            int revision = revisionInfo.Value<int>("Revision");
+
+            try
+            {
```

Comment:
Od této části je kód společný pro Emails / Documents, tak na to pls udělej metodu, třeba DownloadFile.